Sources/Utility/ArgumentParser.swift

```diff
@@ -472,46 +478,61 @@ public final class ArgumentParser: ArgumentParserProtocol {
         }
 
         /// Prints an argument on a stream if it has usage.
-        func print(formatted argument: AnyArgument, on stream: OutputByteStream) {
-            // Don't do anything if we don't have usage for this argument.
-            guard let usage = argument.usage else { return }
+        func print(formatted argument: String, usage: String, on stream: OutputByteStream) {
             // Start with a new line and add some padding.
             stream <<< "\n" <<< space(n: padding)
-            let count = argument.name.characters.count
+            let count = argument.characters.count
             // If the argument name is more than the set width take the whole
             // line for it, otherwise we can fit everything in one line.
             if count >= maxWidth - padding {
-                stream <<< argument.name <<< "\n"
+                stream <<< argument <<< "\n"
                 // Align full width because we on a new line.
                 stream <<< space(n: maxWidth + padding)
             } else {
-                stream <<< argument.name
+                stream <<< argument
                 // Align to the remaining empty space we have.
                 stream <<< space(n: maxWidth - count)
             }
             stream <<< usage
         }
 
-        stream <<< "OVERVIEW: " <<< overview <<< "\n\n"
-        // Get the binary name from command line arguments.
-        let defaultCommandName = CommandLine.arguments[0].components(separatedBy: "/").last!
-        stream <<< "USAGE: " <<< (commandName ?? defaultCommandName) <<< " " <<< usage
+        stream <<< "OVERVIEW: " <<< overview
 
-        if positionalArgs.count > 0 {
+        // We only print command usage for top level parsers.
+        if !isSubparser {
             stream <<< "\n\n"
-            stream <<< "COMMANDS:"
-            for argument in positionalArgs {
-                print(formatted: argument, on: stream)
-            }
+            // Get the binary name from command line arguments.
+            let defaultCommandName = CommandLine.arguments[0].components(separatedBy: "/").last!
```

Comment:
I'm sorry, I don't understand?  Either of what?

What I mean is just instead of splitting components by "/", you could consider arg 0 to be a relative path (we know that it's not always an absolute path) and ask for its last path component.  Just a thought.